CIP-0161/README.md

```diff
@@ -0,0 +1,1871 @@
+
+In these fallback situations, the network still relies on Praos’ guarantees—precisely where Phalanx becomes relevant as a **complementary defense layer**. Phalanx ensures that even when Peras cannot certify blocks, the protocol still benefits from:  
+- **Stronger protection against grinding attacks**, and  
+- **Faster settlement** compared to unmodified Praos.  
+
+Together, they form a **complementary pair**:  
+- **Peras** provides speed when conditions are favorable.  
+- **Phalanx** ensures resilience and strong security guarantees in all cases.
+
+### Technical Depth
+
+The remainder of this document provides the full technical specification for node implementors and researchers. Because Cardano’s security is grounded in **cryptography, probability, and statistical guarantees**, understanding the full details of this proposal requires technical knowledge in these areas. The complete specification is therefore dense: it describes mathematical models, cryptographic primitives, and rigorous proofs to ensure the system can be trusted at scale. Readers interested only in the high-level motivation and community impact may stop here.
+
+Please refer to the CPD "[Ouroboros Randomness Generation Sub-Protocol – The Coin-Flipping Problem](https://github.com/cardano-foundation/CIPs/tree/master/CPS-0021/CPD/README.md)" for a detailed understanding of **randomness generation, leader election in Praos, and the coin-flipping dilemma in consensus protocols**. Moving forward, we will **dive into the core details**, assuming you have the **relevant background** to understand the proposal.
+
+
+## Specification
+
+The core principle of the proposed protocol change is to **substantially escalate the computational cost of each grinding attempt for an adversary**. To achieve this, every honest participant is required to perform a designated computation for each block they produce over an epoch (**432,000 slots - 5 days**). Consequently, an adversary attempting a grinding attack must **recompute these operations for every single attempt**, while being **constrained by the grinding window**, which dramatically increases the resource expenditure. By enforcing this computational burden, we **drastically reduce the feasible number of grinding attempts** an adversary with a fixed resource budget can execute, making randomness manipulation **more expensive and significantly less practical**.
+ 
+### 1. High-Level Overview 
+
+#### 1.1. Changes Relative to Praos
+
+In **Phalanx** , the randomness generation and leader election flows are modified as follows:
+
+![alt text](./image/Praos-vs-Phalanx-Highl-Level.png)
+
+1. The **stake distribution stabilization phase** is shifted **back by one epoch :** The **active** **stake distribution** $`\mathbf{SD}_e`$ used for leader election is now derived from the **end of $epoch_\text{e-3}$** instead of **$epoch_\text{e-2}$**  as in the original Praos protocol.  
```

Comment:
@bwbush, just to be certain, how would such a change impact Leios?

Given that both Leios and Phalanx have been deemed secure additions to Proas, is their combination also considered?